File: c_plus/Remove_Duplicates_from_Sorted_Array.cpp
# include <iostream>
# include <vector>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        int pos = 0;
        for (int i = 1; i < nums.size(); i++)
            if (nums[pos] == nums[i]) 
                continue;
            else 
                nums[++pos] = nums[i];
        return pos + 1;
    }
};

int main() {
    Solution s;
    vector<int> numbers = {0,0,1,1,1,2,2,3,3,4};
    int ans = s.removeDuplicates(numbers);
    cout << ans << endl;
}

File: py/Remove_Duplicates_from_Sorted_Array.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        left = 0
        for i in range(1, len(nums)):
            if nums[left] == nums[i]:
                continue
            else:
                left += 1
                nums[left] = nums[i]
            print(nums)
        return left + 1

if __name__ == '__main__':
    s = Solution()
    print(s.removeDuplicates([1,1,2]))